### DISCUSSION TITLE
prism navigation fails on start up

### DISCUSSION THREAD

**Discussion by stevensrf11** [PrismLibrary/Prism#2604] [WPF]
PRISM Navigation at start up fails
I am not sure this where to ask this question
I am using PRISM 8.0 cire
At startup I try to navigate to a view to a region and it fails.
I the navigation right after the InitializeComponent method

 public partial class MainWindow : ThemedWindow
 {
     public MainWindow()
     {
         InitializeComponent();
         var mv = DataContext as MainWindowViewModel;
         mv.NavigateToToolbar();
         ;
     }

public void NavigateToToolbar()
{
_regionManager.RequestNavigate(RegionNames.NavigationBarRegion, "ToolbarWindow", NavigationCompleted);
}

     private void NavigationCompleted(NavigationResult result)
     {
         var a = 1;
     }

**Comment by brianlagunas** [comment]
You can't navigate to a region that doesn't exist. You have to make sure your view is loaded first.

**Comment by stevensrf11** [comment]
The region exist in the mainwindow.

Now was created were I called the requestnavigate method  I guess it was not

Everthing in the code is just like where you would click a button , handle the button command in the mainview window and   call the requestnavigate method there.

I would like to navigate at startup.

The only place I thought the best place to do this would be would be  after the  InitComposition in the MainWindow Constructor.

So are you saying I should call the requestnavigate method in the loaded event on the MainWindow.

 

 
 

Sent: Wednesday, October 27, 2021 at 2:55 PM

**Comment by stevensrf11** [comment]
I just inject view into to the region from the module at startup thanks

 
 

Sent: Wednesday, October 27, 2021 at 2:55 PM
